interfaces/Portalicious/src/app/pages/project/project-team/project-team.component.ts

```diff
@@ -56,19 +54,19 @@ export class ProjectTeamComponent {
   // this is injected by the router
   projectId = input.required<number>();
 
-  project = injectQuery(this.apiService.getProject(this.projectId));
-  projectUsers = injectQuery(this.apiService.getProjectUsers(this.projectId));
+  project = injectQuery(this.projectApiService.getProject(this.projectId));
+  projectUsers = injectQuery(
+    this.projectApiService.getProjectUsers(this.projectId),
+  );
 
   removeUserMutation = injectMutation(() => ({
     mutationFn: ({ userId }: { userId: number }) =>
-      this.apiService.removeProjectUser(this.projectId(), userId),
+      this.projectApiService.removeProjectUser(this.projectId, userId),
     onSuccess: () => {
       this.toastService.showToast({
         detail: $localize`User removed`,
       });
-      void this.queryClient.invalidateQueries({
-        queryKey: this.apiService.getProject(this.projectId)().queryKey,
-      });
+      void this.projectApiService.invalidateCache(this.projectId);
```

Comment:
There is currently no way to invalidate the "project-users"-query, right?
Because now its not 'tied' to the endpoint/request specifically.

Here, both the queries `projectApiService.getProject()` and `projectApiService.getProjectUsers()` should be invalidated (separately), right?
(Or is the invalidation of 1 cascading to invalidate the other?)

Of should this be tackled by amending the functionality of `invalidateCache` to include an aditional `path`-parameter and calling `invalidateCache(projectId, 'users');` (Or similar)...

Reply:
> There is currently no way to invalidate the "project-users"-query, right?

This is still possible, if necessary, with something like this:

```ts
      void this.queryClient.invalidateQueries({
        queryKey: this.apiService.getProjectUsers(this.projectId)().queryKey,
      });
 ```
 
> Here, both the queries projectApiService.getProject() and projectApiService.getProjectUsers() should be invalidated (separately), right?

Right, but not separately.

> (Or is the invalidation of 1 cascading to invalidate the other?)

Kinda. The way I've currently built it, all of the queries that rely on the projectId will be invalidated when `invalidateCache(projectId)` is used, because they all share the same "beginning of the queryKey array" AKA `['programs', projectId()]`

> Of should this be tackled by amending the functionality of invalidateCache to include an aditional path-parameter and calling invalidateCache(projectId, 'users'); (Or similar)...

I don't think that's necessary (now/yet/potentially ever)

Reply:
Ah, the invalidation happens in a "namespaced" way, not on a simple key-value matching only...
https://tanstack.com/query/latest/docs/framework/angular/guides/query-invalidation#query-matching-with-invalidatequeries

So in this use-case "a bit too much" gets invalidated (buts is saver to be on that side then using data that's 'potentially outdated', in this case.)...

Potentially this approach could be a bit harmful; For example that it would also invalidate a (heavy) query for `programs/1/transactions` (or something unrelated to `users`)..
But lets see when we run into some of those 'inefficiencies' in real life. ;)

Reply:
> So in this use-case "a bit too much" gets invalidated (buts is saver to be on that side then using data that's 'potentially outdated', in this case.)...

Yes, exactly.

> Potentially this approach could be a bit harmful; For example that it would also invalidate a (heavy) query for programs/1/transactions (or something unrelated to users)..

I think there is a potential for telling angular-query to only allow a specific query to be invalidated with an exact match rather than by namespace - but I'm not entirely sure.

Reply:
There's a `exact: true`-parameter I think, looking at that documentation-page linked above...